pkgs/games/gl-gsync-demo/default.nix

```diff
@@ -0,0 +1,43 @@
+{ stdenv, fetchFromGitHub, libX11, freeglut, glew, libXNVCtrl, libXext, lib }:
+
+let
+  pname = "gl-gsync-demo";
+
+  version = "2020-12-27";
+
+  src = fetchFromGitHub {
+    owner = "dahenry";
+    repo = "gl-gsync-demo";
+    rev = "4fd963a8ad880dc2d846394c8c80b2091a119591";
+    sha256 = "1innsmpsd9n9ih80v16rhj2ijrl28jd7x6a4jmxdirba7mjibm8d";
+  };
+in
+
+stdenv.mkDerivation {
+  inherit pname version src;
```

Comment:
```suggestion
stdenv.mkDerivation rec {
  pname = "gl-gsync-demo";
  version = "2020-12-27";
  
  src = fetchFromGitHub {
    owner = "dahenry";
    repo = "gl-gsync-demo";
    rev = "4fd963a8ad880dc2d846394c8c80b2091a119591";
    sha256 = "1innsmpsd9n9ih80v16rhj2ijrl28jd7x6a4jmxdirba7mjibm8d";
  };
```

Reply:
As I said, I prefer the let ... in ... inherit pattern. See https://nix.dev/anti-patterns/language.html#rec-expression.

Reply:
The guide you linked is not the official guide and this does not apply here. Please use rec like many other packages.

Reply:
There is no official guide on this, it's 100% personal preference.

Reply:
Please change it anyway. Rec is cleaner and almost all packages are done like this.

Reply:
> There is no official guide on this, it's 100% personal preference.

Let ins are for new packages are generally only used in two cases: you can't use rec because you need to eval something recursively and when you are inputting other fetchers. Please follow that and change this. Other than this PR is fine and could be merged.

Reply:
Looks like only `src` needs the rec/let. Pname and version should be fine inside the mkDerivation.

Side question: is there a memory or garbage-collection difference between `rec` and `let..in`?

Reply:
It's is about that let in inherit uses more lines and it can create evaluation loops. If you have such loop or need to define a lot of intermediate variables then let in should be used.